### ISSUE TITLE
flexibleBlockPlacementRotation Not work properly 

### ISSUE THREAD

**Issue by santobollove** [maruohon/tweakeroo#111]
Flexible block placement rotation's left , right and middle rotation work like a charm but front and back one always faced at player for clear example check this video.
https://youtu.be/vYcMoOrqjks

**Comment by maruohon** [comment]
Seems to work for me in single player at least.
https://streamable.com/zq0oeh

Are you on a server, and if so, does it have CarpetExtra with the `accurateBlockPlacement` rule enabled?

**Comment by santobollove** [comment]
yes with CarpetExtra  `accurateBlockPlacement` rule enabled it's work in single player world but not in multi player [I tested that in paper server] I think that if left right and middle one work then why back one not working that's why I issued that, Is it possible to fix that issue ??

**Comment by maruohon** [comment]
I don't know why only one of them wouldn't work. But I would have assumed that none of those would work on a Paper server. As a general rule if you are on a server you will need a mod on the server to handle the requested rotations. In single player Tweakeroo handles it itself (you do need to enable the carpet protocol option in any case).

**Comment by Copyablemite4** [comment]
I seem to be having the same issue but mine will not rotate at all. I'm on a server which I am owner of. I see something mentioned about Carpet Extra, I am not aware of what that is and I don't have it on my server. I don't know if that is my issue but just looking for some help to fix it.

**Comment by maruohon** [comment]
@Copyablemite4 CarpetExtra is an extension mod to the base FabricCarpet mod. One of the features in CarpetExtra is support for the so-called "accurate block placement protocol", which basically makes the server rotate the blocks the player is placing, if the client requests a certain modified rotation for the block.

In the current version of Tweakeroo, without this server-side support only blocks that only care about the block side that you click on will be rotated using the Flexible Block Placement rotation feature. This includes blocks such as Hoppers, wood logs, Quartz Pillars. There will likely be a fall-back option for spoofing the player's rotation in a future version of Tweakeroo (after I have the time to rewrite all the placement code), which should allow all or at least most blocks to be rotated without server-side mod support, but that will require some delays in the block placement.

So currently if you want the rotations to work for most blocks, you would have to be running (or able to switch to running) a Fabric server. If you don't already run the FabricCarpet mod, then I would recommend the QuickCarpet mod instead of FabricCarpet and CarpetExtra. That way you basically get the block rotation support with a single mod that has a bit less other extra stuff, and also the block rotation support is better in QuickCarpet. Downloads for that are here: https://github.com/DeadlyMC/QuickCarpet114/releases (expand the Assets under the latest release).

**Comment by Copyablemite4** [comment]
@maruohon I am currently running on paper rn and would prefer to stay on that because of many issues I have run into on other server types, so to get top the point lol, will quick carpet work on paper?

**Comment by maruohon** [comment]
Nope. On any Bukkit variants such as Paper you are stuck either waiting for the player rotation spoofing in Tweakeroo (which I'm not sure if it will even work properly, let alone on Paper), or waiting for someone to make a plugin to support the accurate placement protocol.

**Comment by EltoCode** [comment]
commenting on a closed issue, but is this a known issue for 1.16.5? or perhaps mod conflict? I have the problem were the block looks like it placed correctly before switching back to its different state (If I select it to be placed facing me, it will first place facing me then change to facing the right. It doesn't change to face it's vanilla state). I have carpet and carpet extra with accurate block placement set to true. I even tried the unofficial quick carpet as a substitute but no cigar. 

Running fabric server with lithium.

mod versions: 
```
tweakeroo-fabric-1.16.5-0.10.0-dev.20210917.191839
fabric-carpet-1.16.5-1.4.44+v210714
carpet-extra-1.16.5-1.4.43
------------------------
quickcarpet-3.1.1+1.16.5
```

**Comment by maruohon** [comment]
@EltoCode I don't remember at all what the situation was in 1.16.5. But I'm pretty sure it worked via *some* server side implementation/mod. To clarify, you had the `accurateBlockPlacement` rule (either via CarpetExtra or DeadlyMC/skyrising's QuickCarpet mod) enabled **on the server side**, right?

The fact that it snaps back to a non-vanilla orientation though would point to some other issue. Check that you don't have the After Clicker tweak enabled, it can mess with the rotations in the current implementation, at least with some blocks (Repeaters)... Also the flexible offset combined with a rotation is known to be broken in some cases/orientations.
